src/App.js

```diff
@@ -22,6 +23,7 @@ function App() {
         <Route path="/hacks" component={HacksPage} />
         <Route exact path="/hack/:id" component={HackDetails} />
         <Route path="/hack/:id/edit" component={EditHack} />
+        <Route path="/hack/:id/submit" component={SubmitHack} />
```

Comment:
Will the hackDetails have been loaded if i haven't already visited `/hack/:id` ? (doesnt look like it).

You might want to nest you routes so you have `HackDetails match anything on `/hack/:id*`, it loads the hack and then has a `Switch` in there on `/`, `/edit`, `/submit` that renders the correct component to view, edit or submit.

Reply:
I've moved it into a switch, I think in the future I'd like to move the hack details into a different component to clean up the HackDetails.js file